signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
A possible tiny improvement: call the three options `reject`, `acceptableSignedIdentity`, `signatureForDigest`.  Then it would be clearer what to expect.

`acceptableSignedIdentity` semantics can be restated another way: for any $tag which is admissible, i.e. when policy allows using $repo:$tag, the user can always use $repo@digest-of-that-$tag (to refer to a specific image tagged as $tag in some point in history). If $tag were rejected by the policy, $repo@digest-of-that-$tag will be rejected likewise.

…but then, `signatureForDigest`, really? I am very uncertain that this would ever be needed. (And, after all, we can always add options later; removing something we ship would be worse.)